pkg/network/bridge/lister.go

```diff
@@ -15,26 +20,61 @@ const (
 	maxBridgeNameLength = 12
 )
 
+func getBridgeList(configMap *v1.ConfigMap) dpm.PluginNameList {
+	var plugins = make(dpm.PluginNameList, 0)
+
+	bridges := strings.Split(configMap.Data["bridges"], ",")
+
+	for _, bridgeName := range bridges {
+		if len(bridgeName) > maxBridgeNameLength {
+			glog.Fatalf("Bridge name (%s) cannot be longer than %d characters", bridgeName, maxBridgeNameLength)
+		}
+		plugins = append(plugins, bridgeName)
+	}
+
+	return plugins
+}
+
 type BridgeLister struct{}
 
 func (bl BridgeLister) GetResourceNamespace() string {
 	return resourceNamespace
 }
 
 func (bl BridgeLister) Discover(pluginListCh chan dpm.PluginNameList) {
-	var plugins = make(dpm.PluginNameList, 0)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Fatalf(err.Error())
+	}
 
-	bridgesListRaw := os.Getenv(BridgesListEnvironmentVariable)
-	bridges := strings.Split(bridgesListRaw, ",")
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf(err.Error())
 
-	for _, bridgeName := range bridges {
-		if len(bridgeName) > maxBridgeNameLength {
-			glog.Fatalf("Bridge name (%s) cannot be longer than %d characters", bridgeName, maxBridgeNameLength)
-		}
-		plugins = append(plugins, bridgeName)
 	}
 
-	pluginListCh <- plugins
+	watchlist := cache.NewListWatchFromClient(clientset.CoreV1().RESTClient(), string(v1.ResourceConfigMaps), v1.NamespaceAll, fields.Everything())
+
+	_, informer := cache.NewInformer(watchlist, &v1.ConfigMap{}, 0, cache.ResourceEventHandlerFuncs{
```

Comment:
do we want to handle map deletion?

Reply:
In case a user removes original map and creates a new one, it would be beneficial not to remove it. If a user wants to expose zero bridges, just set empty list. I think it may work better like this, not sure.

Reply:
sure. but I think we should still support that.
i mean we will have 2 events, once deleting the map, and another one creating it.

Reply:
Ok, I'm not against it. Let's add delete callback too.

Reply:
@gageorsburn ^^

Reply:
@phoracek Yup doing it now actually. Sorry I had to put out some unrelated dayjob fires.

Reply:
@gageorsburn no rush :)

Reply:
@phoracek @yuvalif  Delete callback is implemented now. I really like how it flows now actually.